sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -366,14 +375,138 @@ CosmosAsyncContainer setId(String id) {
     }
 
     /**
-     * Gets a CosmosAsyncItem object without making a service call
+     * Reads an item.
+     * <p>
+     * After subscription the operation will be performed.
+     * The {@link Mono} upon successful completion will contain a cosmos item response with the read item
+     * In case of failure the {@link Mono} will error.
+     *
+     * @param itemId the item id
+     * @param partitionKey the partition key
+     * @return an {@link Mono} containing the cosmos item response with the read item or an error
+     */
+    public <T> Mono<CosmosAsyncItemResponse<T>>  readItem(String itemId, PartitionKey partitionKey, Class<T> itemType) {
+        return readItem(itemId, partitionKey, new CosmosItemRequestOptions(partitionKey), itemType);
+    }
+
+    /**
+     * Reads an item.
+     * <p>
+     * After subscription the operation will be performed.
+     * The {@link Mono} upon successful completion will contain a cosmos item response with the read item
+     * In case of failure the {@link Mono} will error.
      *
-     * @param id id of the item
+     * @param itemId the item id
      * @param partitionKey the partition key
-     * @return a cosmos item
+     * @param options the request cosmosItemRequestOptions
+     * @return an {@link Mono} containing the cosmos item response with the read item or an error
      */
-    public CosmosAsyncItem getItem(String id, Object partitionKey) {
-        return new CosmosAsyncItem(id, partitionKey, this);
+    public <T> Mono<CosmosAsyncItemResponse<T>>  readItem(String itemId, PartitionKey partitionKey, 
+                                                  CosmosItemRequestOptions options, Class<T> itemType) {
```

Comment:
passing partition key has ambiguity.

1. partition key can be passed as the second param to readItem
2. CosmosItemRequestOptions has support for `setPartitionKey()`.

if we are always requiring partition key to be passed as a standalone param then the support for setting partition key in `CosmosItemRequestOptions` has to be removed to avoid confusion on public surface area.

Reply:
Forgot to remove the setPartitionKey(). Made it package private now.